node/actors/control_api.go

```diff
@@ -181,62 +180,198 @@ func (api *ControlAPI) subscribe() error {
+
+	workloadResponse, ok := response.(*actorproto.NodeInfo)
 	if !ok {
 		api.logger.Error("Workload listing response from agent supervisor was not the correct type")
 		respondEnvelope(m, InfoResponseType, 500, nil, "Agent supervisor returned the wrong data type")
 		return
 	}
 
-	// TODO: This struct will be replaced by a generated struct from a JSON schema in api/nodecontrol
-	info := struct {
-		NodeID    string                        `json:"node_id"`
-		Workloads []*actorproto.WorkloadSummary `json:"workloads"` // NOTE: the json schema gernerated version of this won't use actorproto
-	}{
-		NodeID:    "Nxxx",
-		Workloads: workloadResponse.Workloads,
-	}
-	respondEnvelope(m, InfoResponseType, 200, info, "")
+	respondEnvelope(m, InfoResponseType, 200, infoResponseFromProto(workloadResponse), "")
 }
 
 func (api *ControlAPI) handleLameDuck(m *nats.Msg) {
-	// TODO
+	ctx := context.Background()
+	_, agentSuper, err := api.self.ActorSystem().ActorOf(ctx, AgentSupervisorActorName)
+	if err != nil {
+		api.logger.Error("Failed to locate agent supervisor actor", slog.Any("error", err))
+		respondEnvelope(m, LameDuckResponseType, 500, nil, fmt.Sprintf("failed to locate agent supervisor actor: %s", err))
+		return
+	}
+
+	response, err := api.self.Ask(ctx, agentSuper, new(actorproto.SetLameDuck))
+	if err != nil {
+		api.logger.Error("Failed to put node in lame duck mode", slog.Any("error", err))
+		respondEnvelope(m, InfoResponseType, 500, nil, fmt.Sprintf("Failed to put node in lame duck mode: %s", err))
+		return
+	}
+
+	workloadResponse, ok := response.(*actorproto.LameDuckResponse)
+	if !ok {
+		api.logger.Error("LameDuck response from agent supervisor was not the correct type")
+		respondEnvelope(m, LameDuckResponseType, 500, nil, "LameDuck response from agent supervisor was not the correct type")
+		return
+	}
+
+	respondEnvelope(m, LameDuckResponseType, 200, lameDuckResponseFromProto(workloadResponse), "")
 }
 
 func (api *ControlAPI) handlePing(m *nats.Msg) {
-	// TODO
-}
+	pingResponse, err := api.nodePingResponse()
+	if err != nil {
+		api.logger.Error("failed to ping node", slog.Any("error", err))
+		respondEnvelope(m, LameDuckResponseType, 500, nil, "failed to ping node")
+		return
+	}
 
-func (api *ControlAPI) handleUndeploy(m *nats.Msg) {
-	// TODO
+	respondEnvelope(m, PingResponseType, 200, pingResponseFromProto(pingResponse), "")
 }
 
-func (api *ControlAPI) handleWorkloadPing(m *nats.Msg) {
-	// TODO
+func (api *ControlAPI) handleAgentPing(m *nats.Msg) {
+	req := new(nodecontrol.AgentPingResponseJson)
+	err := json.Unmarshal(m.Data, req)
+	if err != nil {
+		api.logger.Error("Failed to unmarshal agent ping request", slog.Any("error", err))
+		respondEnvelope(m, AgentPingResponseType, 500, nil, fmt.Sprintf("failed to unmarshal agent ping request: %s", err))
+		return
+	}
+
+	ctx := context.Background()
+	_, agentSuper, err := api.self.ActorSystem().ActorOf(ctx, AgentSupervisorActorName)
```

Comment:
shouldn't we be pulling the agent name/workload type from the `APING` subject, and pinging the specific agent? If that's the case, we should locate the individual agent process and not the agent super. We should only ping the agent super if we want it to aggregate ping responses from all agents.